scripts/gha/build_desktop.py

```diff
@@ -182,6 +182,8 @@ def cmake_configure(build_dir, arch, msvc_runtime_library='static', linux_abi='l
   else:
     # workaround, absl doesn't build without tests enabled
     cmd.append('-DBUILD_TESTING=off')
+  # When building from GitHub Actions, this should always be set.
+  cmd.append('-DFIREBASE_GITHUB_ACTION_BUILD=ON')
```

Comment:
This shouldn't be hard coded here.  Our build scripts are run locally by the team and so local builds will be reported as GHA builds.  Instead create a runtime flag on build_desktop.py that appends this -D command, and then alter `.github/workflows/desltop.yml` and `.github/workflows/cpp-packaging.yml` to add a `--gha_build` flag (or something) when the workflow invokes `build_desktop.py`

Reply:
I think I made the changes you wanted. Take a look and make sure I set it up right.